### DISCUSSION TITLE
https://our.umbraco.com/packages needs a major update

### DISCUSSION THREAD

**Discussion by tormnator** [umbraco/Umbraco.Packages#40] [Ideas] [answered]
In my mind the most important task at the moment is to give the Packages web page(s) a major face lift. Simply put, why spend a lot of time and effort into the package creation/maintenance/documentation process if our users can't find the packages that are already there?

The Packages page should have the following features:

1. SSF - Search, Sort, and Filter. Out of the box, and when displaying search results, we should be able to filter the list, and we should be able to sort the results. A nice to have, but not required feature, would be to search within the search results. Categories to both filter and search on could include:
    * Date of original release
    * Date of latest release
    * Popularity (download count)
    * Rating
    * Title
2. Rating system - the current voting/heart feature is too simplistic and hard to find. A star/rating system would be an improvement
3. Reviews - this is a standard feature in most plugin/app repositories
4. Promotion - ability to showcase certain packages e.g. "top 10 latest, most downloads, etc.", "recommended by Umbraco", etc.
5. The individual package page should be updated and contain:
    * The same sort/filter fields which are used on the search page
    * Reviews
    * The long "Package Compatibility" sidebar should be moved to the bottom to avoid it dominating the available space
    * This is important: what is the package's "footprint"? How does it impact my website installation? Files only? web.config? Database changes? Is it cloud compatible? Are all changes 100% reversible when uninstalled? For me one of the primary reasons for not installing a package is the fear that it will break something on my page and/or that it will block me from performing future updates.

Miscellaneous improvements:

* Every search result page should identify how many packages are listed/were found.
* Every search result should contain the most important information. Typically this would include the sort/filter categories.
* Categories are too broad. For instance, I'm interested in packages related to the media library and images, but am unable to find those easily

**Comment by jmayntzhusen** [comment]
Hey @tormnator 

Thanks a lot for the detailed feedback here, I agree on all of these points, and they could probably be split into individual issues to improve the Our repository bit by bit.

A few follow up questions though:

> Simply put, why spend a lot of time and effort into the package creation/maintenance/documentation process if our users can't find the packages that are already there?

Is this how users find packages? Not through the backoffice?
The backoffice feed is from Our as well, but separate to the Our page, but I guess both should be updated.

> The individual package page should be updated and contain:
> The same sort/filter fields which are used on the search page

Why would you need to sort/filter fields on the individual package page? Not sure I understand what you mean here 🙂

> This is important: what is the package's "footprint"? How does it impact my website installation? Files only? web.config? Database changes? Is it cloud compatible? Are all changes 100% reversible when uninstalled? For me one of the primary reasons for not installing a package is the fear that it will break something on my page and/or that it will block me from performing future updates.

This one is a bit tough, because as far as I know there are 3 main methods people install packages:
1. NuGet package
1. Directly from the Umbraco backoffice
1. Uploading a ZIP in the Umbraco backoffice (could be from Our or somewhere else)

Listing this sort of information would only help people using method 3, whereas informing when you try to install a package in the backoffice would cover 2 + 3. NuGet we can't really do anything about.


Finally I also just want to say that while these are all very good points, and something we want to do it is also something anyone can help out with. Our and Umbraco are both open source, so if you want to improve the package feeds you can by submitting PRs
https://github.com/umbraco/OurUmbraco 
https://github.com/umbraco/Umbraco-CMS

**Comment by dawoe** [comment]
Hi @jmayntzhusen 

I don't know a lot of people who actually install packages through the backoffice. For people working with source control and CI/CD servers will probably use nuget. And if nuget it's not an option it is problaly a local install by downloading the packge from our. So they can put everything in source control.

I actually can't remember when I last used the backoffice to install a package. 

Maybe this info can be retreived from the download stats from our. Take a popular package and see how downloads on our compares to installs through the backoffie

Dave

**Comment by jmayntzhusen** [comment]
Hmm that's a good point @dawoe 

And if it was on Our we could pass it along to the backoffice anyways and make sure it is displayed when you click on a package..

**Comment by LottePitcher** [comment]
I discover and research packages from the Our website.  I then install packages via Nuget, assuming there is one of course.  I think I avoid 'discovering' packages via the back office so that I don't accidentally install from there (when I should be doing via Nuget for source control/build reasons as Dave says).  Also I'm not always logged into a back office of Umbraco, so it's easier to visit Our. 

Improving the discover-ability of packages on Our would also be of benefit to people doing an evaluation of the Umbraco product (i.e. to see what 'gaps' in the CMS functionality have been met via packages).

**Comment by tormnator** [comment]
Hi @jmayntzhusen,

Reples to your questions:

> Is this how users find packages?

- I can only answer for myself here, and so far I have only used the packages web page on Our, almost never via the backoffice page.

> Why would you need to sort/filter fields on the individual package page?

- I didn't mean you need to sort/filter on the fields there, just that the fields need to be visible on the page :-)

> This one is a bit tough, because as far as I know there are 3 main methods people install packages: ...

- Having multiple ways of installing packages is not ideal. Ideally all three methods/sites should use the same underlying package information data(base) so that at least they stay in sync. The Backoffice and the Our packages page should just be different (hopefully not too different) views of the same data. The Nuget pages should link to the Our package page for information.

I've noticed another weakness with Nuget vs. Backoffice installation: in the Backoffice, the "installed packages" page doesn't reflect the installed version numbers of any Nuget-installed packages. So I have to check in 2 places to be sure of which version I have installed.

Contributing to the Umbraco projects is on my bucket list, but unfortunately I don't have much time for that these days.

-Tor

**Comment by jmayntzhusen** [comment]
Hey @tormnator 

We have started looking into Our improvements. I have added this issue to our project board as "Needs scoping". 
We will take a look at this issue on our next package team meeting and discuss what we want to do about this, then likely split it into more concrete issues.

**Comment by ravimotha** [comment]
Hi, I was just looking at the current site and i realised the categories are not really "friendly"

ie 
what does collaboration mean
backoffice solutions-  does this need to be more specific 
by back office section
developer tools
should we have content apps as a cotegory?
Umbraco PRO
Website Utilities
Umbraco Cloud
Starter Kits

**Comment by ronaldbarendse** [comment, marked as answer]
Most of these suggestions have already been included when developing the new [Marketplace site](https://marketplace.umbraco.com/):
- Search, sort and filter: you can now search, add additional filters and change the sorting.
- Rating system: this is something we discussed, but not something we want to currently implement (as it would also require administration/management, spam prevention, etc.). The 'most popular' overview is based on most active installs, so that already gives a hint of what packages are widely used.
- Reviews: for the same reasons as the rating system, we've decided to not implement this for now. We also don't want to add another place for package owners to manage feedback, since most packages will already have an issue tracker/discussion board.
- Promotion: we highlight Umbraco HQ products, partner packages and the package view shows any awards/nominations it received, besides having the 'most popular' overview.

Thanks again for sharing your feedback! If you have more suggestions and/or want to continue discussing them, please create a new issue on https://github.com/umbraco/Umbraco.Marketplace.Issues.
